js/popupGallery.js

```diff
@@ -0,0 +1,74 @@
+import {picturesContainer} from './renderingPicture.js';
+
+const displayingWindow = document.querySelector('.big-picture');
+const closeBigPhoto = displayingWindow.querySelector('.big-picture__cancel');
+const bigPicture = displayingWindow.querySelector('.big-picture__img img');
+const likesCounter = displayingWindow.querySelector('.likes-count');
+const commentCount = displayingWindow.querySelector('.comments-count');
+const commentsList = displayingWindow.querySelectorAll('.social__comments li');
+const descriptionPhoto = displayingWindow.querySelector('.social__caption');
+
+const addClass = ()=>{
+  const socialCommentCount = displayingWindow.querySelector('.social__comment-count');
+  const commentsLoader = displayingWindow.querySelector('.comments-loader');
+  document.querySelector('body').classList.add('modal-open');
+  socialCommentCount.classList.add('hidden');
+  commentsLoader.classList.add('hidden');
+};
+
+const closePopup = ()=>{
+  event.preventDefault();
+  displayingWindow.classList.add('hidden');
+  closeBigPhoto.removeEventListener('click',closePopup);
+};
+const closePopupKey = ()=>{
+  event.preventDefault();
+  if(event.key === 'Escape'){
+    displayingWindow.classList.add('hidden');
+    closeBigPhoto.removeEventListener('click', closePopup);
+    document.addEventListener('keydown',closePopupKey);
+  }
+};
+const showPopup = ()=>{
```

Comment:
`addComments` и `setPhotoData` нужно перенести выше этой функции
```suggestion
const showPopup = (event) => {
  event.preventDefault();
  const photoId = parseInt(event.target.closest('[data-id]')?.dataset.id, 10);

  if (!photoId) {
    return;
  }

  const photoData = photos.find((photo) => photo.id === photoId);
  if (!photoData) {
    return;
  }
  
  setPhotoData(photoData);

  displayingWindow.classList.remove('hidden');
  closeBigPhoto.addEventListener('click', hidePopup);
  document.addEventListener('keydown', hidePopup);
  document.body.classList.add('modal-open');
};
```